File: Spring-Boot/jpa-udemy/src/main/java/dev/vitorvidal/jpaudemy/entity/Course.java
package dev.vitorvidal.jpaudemy.entity;

import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "course")
@NamedQuery(name = "query_get_all_courses", query = "Select c From Course c")
public class Course {
    @Id
    @GeneratedValue
    private Long id;
    @Column(name = "name")
    private String name;
    @OneToMany(mappedBy = "course", fetch = FetchType.EAGER)
    private final List<Review> reviews = new ArrayList<>();
    @UpdateTimestamp
    private LocalDateTime lastUpdatedDate;
    @CreationTimestamp
    private LocalDateTime createdDate;

    protected Course() {
    }

    public Course(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public List<Review> getReviews() {
        return reviews;
    }

    public void addReviews(Review review) {
        this.reviews.add(review);
    }

    public void removeReviews(Review review) {
        this.reviews.remove(review);
    }

    @Override
    public String toString() {
        return "Course{" + "id=" + id + ", name='" + name + '\'' + '}';
    }
}

File: Spring-Boot/jpa-kotlin-example/src/main/kotlin/dev/vitorvidal/jpakotlinexample/entity/Course.kt
package dev.vitorvidal.jpakotlinexample.entity

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "course")
@NamedQuery(name = "query_get_all_courses", query = "Select c From Course c")
class Course {
    @Id
    @GeneratedValue
    var id: Long? = null

    @Column(name = "name")
    var name: String? = null

    @UpdateTimestamp
    val lastUpdatedDate: LocalDateTime? = null

    @CreationTimestamp
    val createdDate: LocalDateTime? = null

    protected constructor()
    constructor(name: String?) {
        this.name = name
    }

    override fun toString(): String {
        return "Course{id=$id, name='$name'}"
    }
}